src/main/java/appeng/core/features/registries/P2PTunnelRegistry.java

```diff
@@ -146,7 +146,7 @@ public TunnelType getTunnelTypeByItem( final ItemStack trigger )
 					return this.tunnels.get( is );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) appeng.core.features.registries.P2PTunnelRegistry.getTunnelTypeByItem(ItemStack) makes inefficient use of keySet iterator instead of entrySet iterator [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=findbugs%3AWMI_WRONG_MAP_ITERATOR)